src/server/system_services/master_key_manager.js

```diff
@@ -95,20 +126,24 @@ class MasterKeysManager {
      * @returns {Object} 
      */
     new_master_key(options) {
-        const { description, master_key_id, cipher_type } = options;
+        const { description, master_key_id, root_key_id, cipher_type } = options;
         const _id = db_client.new_object_id();
+        const master_id = (master_key_id && db_client.parse_object_id(master_key_id)) || undefined;
+        const root_id = root_key_id || master_id === undefined ? this.current_root_key : undefined;
```

Comment:
can you explain when it happens where the ket is not root_key && master_id is undefined?